### PR TITLE
fix security issue with seal

### PR DESCRIPTION
**Pull request by g2flyer**
This fix a major and a minor security issue with seal of contract state and unifies the settings for seal and attestation.

Note this will stop existing contracts from being read as sealkey changes!

### CODE REVIEW

File: build/__tools__/run-tests.sh [hunk 1]
```diff
@@ -186,11 +186,12 @@ try pdo-create --config ${CONFIG_FILE} --ledger ${PDO_LEDGER_URL} \
...
     --contract mock-contract --source _mock-contract.scm
 
-# this will invoke the increment operation 50 times; the objective
+# this will invoke the increment operation n times; the objective
 # of this test is to ensure that the client touches multiple, independent
 # enclave services and pushes missing state correctly
-say increment the value with a simple expression
-for v in $(seq 1 50) ; do
+declare -i n=50
+say increment the value with a simple expression ${n} times
```
Comment (cmickeyb) [inline build/__tools__/run-tests.sh]: can i suggest that if we are going to change this... its really better to do it right. Given that the point of doing multiple iterations is just to ensure that multiple eservices are contacted, why not just do a round robin on the eservice urls.

File: build/__tools__/run-tests.sh [hunk 2]
```diff
@@ -217,7 +218,7 @@ value=$(pdo-update --config ${CONFIG_FILE} --ledger ${PDO_LEDGER_URL} \
                    --logfile __screen__ --loglevel warn \
                    --identity user1 --save-file ${SAVE_FILE} \
                    "'(get-value)")
-if [ $value != "51" ]; then
+if [ $value != $((51)) ]; then
     die contract has the wrong value
```
Comment (cmickeyb) [inline build/__tools__/run-tests.sh]: this should be n+1 if you're gonna stick to this test.

File: common/crypto/sgx_defaults.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/* Copyright 2018 Intel Corporation
+ *
```
Comment (cmickeyb) [inline common/crypto/sgx_defaults.h]: 2019

File: eservice/lib/libpdo_enclave/contract_enclave.cpp [hunk 4]
```diff
@@ -99,6 +99,38 @@ pdo_err_t ecall_ShutdownContractWorker(){
     return result;
 }
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+pdo_err_t ecall_CalculateSealedContractKeySize(
+  size_t contractIdSize,
+  size_t* pSealedContractKeySize)
+{
```
Comment (cmickeyb) [inline eservice/lib/libpdo_enclave/contract_enclave.cpp]: for ecall parameters... could you please use the in/out convention?

File: common/crypto/sgx_defaults.h (hunk 3 above)
Comment (g2flyer) [reply common/crypto/sgx_defaults.h]: good catch!

File: eservice/lib/libpdo_enclave/contract_enclave.cpp (hunk 4 above)
Comment (g2flyer) [reply eservice/lib/libpdo_enclave/contract_enclave.cpp]: should be fixed now

File: build/__tools__/run-tests.sh (hunk 1 above)
Comment (g2flyer) [reply build/__tools__/run-tests.sh]: original motivation was just to have a single place where 50 and related numbers are defined but as you point out below i failed as I must have somehow screwed up applying my old patch.  Anyway, changed it now to round-robin in updated patch.

**Comment by bvavala** [comment]
indentation issues will be solved in a later PR

**Pull request merged**